primitives/io/src/lib.rs

```diff
@@ -87,16 +87,41 @@ pub enum EcdsaVerifyError {
 	BadSignature,
 }
 
+/// This is the initial kill storage result used
 /// The outcome of calling `storage_kill`. Returned value is the number of storage items
 /// removed from the trie from making the `storage_kill` call.
 #[derive(PassByCodec, Encode, Decode)]
-pub enum KillStorageResult {
+enum KillStorageResultPrev {
 	/// No key remains in the child trie.
 	AllRemoved(u32),
+
 	/// At least one key still resides in the child trie due to the supplied limit.
 	SomeRemaining(u32),
 }
 
+/// The outcome of calling `storage_kill`. Returned value is the number of storage items
+/// removed from the trie and also removed from overlay from making the `storage_kill` call.
+#[derive(PassByCodec, Encode, Decode)]
+pub enum KillStorageResult {
+	/// No key remains in the child trie.
```

Comment:
I think we should also fix the doc, this result is not only for child trie but also for main trie
```suggestion
 	/// No key remains in the child trie or at the top trie prefix.
```